### DISCUSSION TITLE
How to signout/destroy session on the server

### DISCUSSION THREAD

**Discussion by Mosquid** [nextauthjs/next-auth#2826] [Help]
What would be an alternative to signOut function on the server? Basically, I'm looking for a way to redirect a user to some page while destroying their session (invalidate cookie?).

**Comment by balazsorban44** [comment]
I don't think you can delete a cookie without actually receiving the `Set-Cookie` response header in the browser. 👀

If you use a database-persisted session though, you can delete it from the database, and next time the user tries to grab their session, it won't be valid anymore.

**Reply by Mosquid** [reply]
Thank you for your answer. 

Yes, I'm aware that it requires sending "set-cookie" and I'm looking for a way to do it right. Here is what I'm trying to achieve:

- getServerSideProps is trying to fetch current user's data from my backend
-  If 401 is returned, I need to terminate the session

Is there a way to achieve this in one redirect?

**Reply by agusterodin** [reply]
> Thank you for your answer.
> 
> Yes, I'm aware that it requires sending "set-cookie" and I'm looking for a way to do it right. Here is what I'm trying to achieve:
> 
> * getServerSideProps is trying to fetch current user's data from my backend
> * If 401 is returned, I need to terminate the session
> 
> Is there a way to achieve this in one redirect?

I am doing very similar thing with fetching user profile from backend in the session callback. Were you able to achieve this?
